crates/ruma-events-macros/src/event_enum.rs

```diff
@@ -1037,3 +1071,40 @@ impl EventEnumEntry {
         Ok(EventEnumVariant { attrs, ident })
     }
 }
+
+pub(crate) fn expand_from_impls_derived(input: DeriveInput) -> TokenStream {
+    let variants = match &input.data {
+        Data::Enum(DataEnum { enum_token: _, brace_token: _, variants }) => variants,
+        _ => panic!("this derive macro only works with enums"),
+    };
+
+    let from_impls = &variants
+        .iter()
+        .map(|variant| match &variant.fields {
+            syn::Fields::Unnamed(fields) => {
+                if fields.unnamed.len() == 1 {
```

Comment:
This branch inside the match arm can be changed to be a match guard, with either the two panics becoming one with a slightly more generic message, or there just being a second `Fields::Unnamed` match arm. That would reduce rightwards (indentation) drift.